### PR TITLE
Improve travis integration

### PR DESCRIPTION
**Pull request by adambabik**
Add linter and divide Travis CI integration into stages.

--- 
I encourage you to check the Travis output. It's possible to immediately see where the problem is. Is it linter, unit tests or maybe e2e tests?

It's also possible to restart a single stage. If only e2e tests failed and you think it's a flake, it's possible to rerun only that single stage.

### CODE REVIEW

**Comment by tiabc** [comment]
Implements #387, right?

**Comment by adambabik** [comment]
@tiabc I didn't notice that :/ Yeah, it implements that and https://github.com/status-im/status-go/issues/284.

**Comment by adambabik** [comment]
> By the way, is that possible to have these stages like what we already have on the github page as "pr", "push" and "branch" things?

I don't know but I guess it's not available yet. I quickly looked around but didn't notice such an option.

**Comment by tiabc** [comment]
@adambabik please, resolve conflicts.
@b00ris please, review.

**Comment by tiabc** [comment]
@adambabik please, resolve conflicts.
@b00ris please, review.

**Comment by adambabik** [comment]
@tiabc fixed conflicts.

**Comment by tiabc** [comment]
Merged develop and slightly modified this PR in favour of #419.
This PR now only tackles build stages as I find running unit tests before e2e tests a nice improvement.

File: .travis.yml [hunk 1]
```diff
@@ -15,8 +18,13 @@ jobs:
   include:
     - stage: lint
       script: make lint
-    - stage: ci
-      script: make ci
+    - stage: Lint
```
Comment (influx6) [inline .travis.yml]: Is this not a duplicate of the previous?

Comment (tiabc) [reply .travis.yml]: Yes, it is, thanks, will fix.

**Comment by tiabc** [comment]
@b00ris only left Travis stages here as the rest's been handled in #419

**Pull request merged**